repos/system_upgrade/el7toel8/actors/checkpcidrivers/actor.py

```diff
@@ -0,0 +1,155 @@
+from itertools import chain
+
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.models import PCIDevices, RestrictedPCIDevices
+from leapp.reporting import Report, create_report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+
+def render_report(
+    restricted_driver_names_on_host,
+    restricted_pci_ids_on_host,
+    inhibit_upgrade=False,
+):
+    """Render the report for restricted PCI devices in use.
+
+    :param restricted_driver_names_on_host: a dict with restricted driver names
+        presented on a host machine
+    :type restricted_driver_names_on_host: Dict[DriverName, DeviceData]
+    :param restricted_pci_ids_on_host: a dict with restricted pci ids
+        presented on a host machine
+    :type restricted_pci_ids_on_host: Dict[PciId, DeviceData]
+    :param inhibit_upgrade: if True, then the report will contain the INHIBITOR
+        flag (this will cause the IPU to be inhibited)
+    :type inhibit_upgrade: bool
+    """
+    unavailable_driver_names = ",\n".join(
+        driver
+        for driver, data in restricted_driver_names_on_host.items()
+        if data["available_rhel8"] == 0
+    )
+    unavailable_pci_ids = ",\n".join(
+        pci_id
+        for pci_id, data in restricted_pci_ids_on_host.items()
+        if data["available_rhel8"] == 0
+    )
+    unsupported_driver_names = ",\n".join(
+        driver
+        for driver, data in restricted_driver_names_on_host.items()
+        if data["supported_rhel8"] == 0
+    )
+    unsupported_pci_ids = ",\n".join(
+        pci_id
+        for pci_id, data in restricted_pci_ids_on_host.items()
+        if data["supported_rhel8"] == 0
+    )
+    title = "Detected PCI drivers that are restricted on RHEL8."
+    summary = (
+        "There are following drivers that are restricted on RHEL 8 system.\n\n"
+        "Host devices which are unavailable on RHEL8:\n {},\n{}\n\n"
+        "Host devices which are unsupported on RHEL8:\n {},\n{}\n\n"
+    ).format(
+        unavailable_driver_names,
+        unavailable_pci_ids,
+        unsupported_driver_names,
+        unsupported_pci_ids,
+    )
+    reports = [
+        reporting.Title(title),
+        reporting.Summary(summary),
+        reporting.Severity(reporting.Severity.HIGH),
+        reporting.Tags([reporting.Tags.KERNEL, reporting.Tags.DRIVERS]),
+    ]
+    if inhibit_upgrade:
+        reports += [reporting.Flags([reporting.Flags.INHIBITOR])]
+    return reports
+
+
+class CheckPCIDrivers(Actor):
+    """Check if host PCI devices drivers has restrictions on target system.
```

Comment:
I will just remind you to use the docstring style we use in our project.

Reply:
done:) you know what I feel man. Loving the python so much and being forced to use atypical docstrings formating, just killing. There are no reasons actually provided, why do we apply an exception

Reply:
I know :) Still, it's not so atypical when it's part of standard too :)